### PR TITLE
Add documentation for '/debug/*' endpoints

### LINKED ISSUE #3174: Document the /debug API endpoints
I think we should document the `/debug/*` API endpoints in https://cortexmetrics.io/docs/api/, including the recently added `/debug/fgprof` #3131.

### PR DESCRIPTION
**Pull request by Jigar3**
Signed-off-by: Jigar <chavada.jigar.3@gmail.com>

**What this PR does**:
- Adds documentation of `/debug/*` endpoints

**Which issue(s) this PR fixes**:
Fixes #3174 

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by Jigar3** [comment]
Created the first draft, let me know if you want me to modify or add anything.

**Comment by Jigar3** [comment]
@pracucci @pstibrany Please could you review this.

File: docs/api/_index.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ For the sake of clarity, in this document we have grouped API endpoints by servi
 | [Remote write](#remote-write) | Distributor | `POST /api/v1/push` |
 | [Tenants stats](#tenants-stats) | Distributor | `GET /distributor/all_user_stats` |
 | [HA tracker status](#ha-tracker-status) | Distributor | `GET /distributor/ha_tracker` |
+| [Pprof](#pprof) | Debugger | `GET /debug/pprof` |
+| [Fgprof](#fgprof) | Debugger | `GET /debug/fgprof` |
```
Comment (pstibrany) [inline docs/api/_index.md:29]: These are exposed by all Cortex services:

```suggestion
| [Pprof](#pprof) | _All services_ | `GET /debug/pprof` |
| [Fgprof](#fgprof) | _All services_ | `GET /debug/fgprof` |
```

**Review by pstibrany — COMMENTED** [review]
I'd change the "Service" part, since it refers to Cortex service, and there is no "Debugger".

File: docs/api/_index.md (hunk 1 above)
Comment (pracucci) [reply docs/api/_index.md:29]: Because of this, please move it above, at the end of the "All services" group.

File: docs/api/_index.md [hunk 2]
```diff
@@ -170,6 +172,28 @@ GET /ha-tracker
 
 Displays a web page with the current status of the HA tracker, including the elected replica for each Prometheus HA cluster.
 
+## Debugger
```
Comment (pracucci) [inline docs/api/_index.md:175]: Move above, grouped together with the "All services"

File: docs/api/_index.md [hunk 3]
```diff
@@ -170,6 +172,28 @@ GET /ha-tracker
...
+
+### Pprof
+
+```
+GET /debug/pprof/*
```
Comment (pracucci) [inline docs/api/_index.md:180]: Could you list all endpoints? The final user may have an hard time figuring out which are supported, even if we linked to pprof doc.

**Review by pracucci — COMMENTED** [review]
Thanks! A couple of nits, and we're good to go. Good job!

File: docs/api/_index.md (hunk 3 above)
Comment (Jigar3) [reply docs/api/_index.md]: Done. Could you check it?

File: docs/api/_index.md (hunk 2 above)
Comment (Jigar3) [reply docs/api/_index.md]: Done

File: docs/api/_index.md (hunk 1 above)
Comment (Jigar3) [reply docs/api/_index.md]: Moved

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Review by pstibrany — APPROVED** [review]
Thank you!

**Pull request merged**